Bash_scripts/HW20/roles/hw201/tasks/main.yml

```diff
@@ -0,0 +1,32 @@
+#For Linux group
+- name: Install git and pyhton3
+  import_tasks: install.yml
+  
+#Install docker
+- name: Install docker
+  import_tasks: docker.yml 
+   
+#Add user for linux group
+- name: Add the user 'hw20' with a sh shell and '/opt/hw20' home directory
+  ansible.builtin.user:
+   name: hw20
+   shell: /bin/sh
+   home: /opt/hw20
+   groups: "{{ group_sudo }}"
+   state: present
+  become: true
+  when: "'linux' in group_names"
+     
+#Reset connection  
+- name: reset ssh connection to allow user changes to affect 'current login user'
+  meta: reset_connection  
+  
+#Pull public git repo
+- name: Pull public git repo as user hw20 to /opt/hw20/
```

Comment:
If you do it via root do not forget to change owner

Reply:
you don't need to use root here at all. deleted.